lib/src/api.dart

```diff
@@ -37,19 +37,16 @@ abstract class Serializer {
   static Object fromJson(String json, [Type type]) => _fromJson(json, type);
 
   /// Convert a Map<String, dynamic>
-  static Object fromMap(Map map, [Type type]) => _fromMap(map, type);
+  static Object fromMap(Map map, [Type type, List<Type> mapOf]) => _fromMap(map, type, mapOf);
```

Comment:
There are no case with more than 2 types.

I prefer to implement the first solution, because `mapKey` and `mapValue` are linked.
```dart
fromMap(Map, [String, DateTime])
```
```dart
fromMap(Map, String, DateTime)
```
What do you think ?

For the moment, `mapOf[0]` must be a `String` for JSON codec. However, with the refactorisation I did, in the future, this package could be easily use with another codec than JSON.